am-lib/src/main/java/uk/gov/hmcts/reform/amlib/DefaultRoleSetupImportService.java

```diff
@@ -0,0 +1,130 @@
+    }
+
+    public void addRole(@NonNull String roleName, RoleType roleType, SecurityClassification securityClassification,
+                        AccessManagementType accessManagementType) {
+        if (roleName.isEmpty()) {
+            throw new IllegalArgumentException("Role name cannot be empty");
+        }
+
+        jdbi.useExtension(DefaultRoleSetupRepository.class,
+            dao -> dao.addRole(roleName, roleType, securityClassification, accessManagementType));
+    }
+
+    public void addResourceDefinition(@NonNull String serviceName,
+                                      @NonNull String resourceType,
+                                      @NonNull String resourceName) {
+        if (serviceName.isEmpty()) {
+            throw new IllegalArgumentException("Service name cannot be empty");
+        }
+
+        if (resourceType.isEmpty() || resourceName.isEmpty()) {
+            throw new IllegalArgumentException("Resource cannot contain empty values");
+        }
+
+        jdbi.useExtension(DefaultRoleSetupRepository.class, dao ->
+            dao.addResourceDefinition(serviceName, resourceType, resourceName));
+
+    }
+
+    public void grantDefaultPermission(DefaultPermissionGrant defaultPermissionGrant) {
+        jdbi.useTransaction((handle) -> {
+            DefaultRoleSetupRepository dao = handle.attach(DefaultRoleSetupRepository.class);
+            try {
+                defaultPermissionGrant.getAttributePermissions().forEach((attribute, permissionAndClassification) -> {
+                    dao.createResourceAttribute(ResourceAttribute.builder()
+                        .serviceName(defaultPermissionGrant.getServiceName())
+                        .resourceName(defaultPermissionGrant.getResourceName())
+                        .resourceType(defaultPermissionGrant.getResourceType())
+                        .attribute(attribute.toString())
+                        .securityClassification(permissionAndClassification.getValue())
+                        .build()
+                    );
+
+                    dao.grantDefaultPermission(
+                        DefaultPermission.builder()
+                            .serviceName(defaultPermissionGrant.getServiceName())
+                            .resourceType(defaultPermissionGrant.getResourceType())
+                            .resourceName(defaultPermissionGrant.getResourceName())
+                            .attribute(attribute.toString())
+                            .roleName(defaultPermissionGrant.getRoleName())
+                            .permissions(Permissions.sumOf(permissionAndClassification.getKey()))
+                            .build());
+                });
+
+            } catch (Exception e) {
+                throw new ErrorAddingEntriesToDatabaseException(e);
+            }
+        });
+    }
+
+    public void truncateAllDefaultPermissionsForService(String serviceName, String resourceType) {
```

Comment:
@swalker125 don't you think `All` is redundant? I have feeling `truncate` is enough.